barretenberg/cpp/docs/src/sumcheck-outline.md

```diff
@@ -0,0 +1,438 @@
+
+Non ZK-Sumcheck Outline {#NonZKSumcheck}
+========
+- - -
+ ### Sumcheck Relation {#SumcheckRelation}
+
+ Given multilinear polynomials \f$ P_1,\ldots, P_N \in \mathbb{F}[X_0,\ldots, X_{d-1}] \f$ and a polynomial \f$ F \f$ in \f$ N \f$ variables, we run Sumcheck over the polynomial
+ \f{align}{
+    \tilde{F}
+    (X_0,\ldots, X_{d-1}) =
+    pow_{\beta}(X_0,\ldots, X_{d-1}) \cdot F\left( P_1 (X_0,\ldots, X_{d-1}), \ldots, P_N (X_0,\ldots, X_{d-1}) \right)
+    \f}
+to establish that \f$ F(P_1(\vec \ell),\ldots, P_N(\vec \ell) ) = 0 \f$, i.e. that \f$ F \f$ is satisfied at every
+point of \f$\{0,1\}^d\f$.
+
+ In the implementation, the relation polynomial \f$ F \f$ is specified by the Flavor.
+
+ ### Main Parameters {#MainParameters}
+
+The following constants are used in this exposition. 
+
+ |     Notation      |           | \f$ \sim \f$   Upper Bound |
+ --------------------|---------------|-----------|
+ | \f$ d \f$         | \ref multivariate_d "number of variables" in multilinear  polynomials \f$ P_1,\ldots, P_N\f$        | \f$ 20 \f$  |
+ | \f$ N \f$         | number of Prover Polynomials specified by <Flavor ::NUM_ALL_ENTITIES>                               | \f$ 60 \f$  | 
+ | \f$ N_w \f$  	 | number of Witness Polynomials specified bt <Flavor ::NUM_WITNESS_ENTITIES>                          | \f$ 17 \f$  | 
+ | \f$ n \f$         | \ref multivariate_n "size of the hypercube", i.e. \f$ 2^d\f$.                                       | \f$ 2^{20} \f$ |
+ | \f$ D \f$         | \ref bb::SumcheckProverRound< Flavor >::BATCHED_RELATION_PARTIAL_LENGTH "maximum partial degree of" \f$\tilde{F}\f$ | \f$ 12 \f$ |
+ | \f$ D_w\f$        | maximum witness degree of \f$ F \f$ | \f$ 5 \f$ |
+
+#### Maximum Witness Degree 
+The significance of this parameter becomes apparent in Section [Masking Evaluations of Multilinear Witnesses](#MaskingEvalsOfWitnesses). It is formally defined as follows 
+\f{align}{
+	D_w = \deg_{P_1, \ldots, P_{N_w}} F(P_1,\ldots, P_{N})
+\f}
+where by \f$ \deg_{P_1, \ldots, P_{N_w}} \f$ we mean the <b> total degree </b> of relation polynomial \f$ F \f$ in the first \f$N_w\f$ variables. 
+
+## Sumcheck Prover Algorithm {#NonZKSumcheckProver}
+- - -
+We remark that prior to running Sumcheck, the prover commits to multilinear polynomials \f$P_1,\ldots, P_{N_w}\f$, and sends the commitments to the verifier and that the total sum and the relation polynomial \f$ \tilde F \f$ are public. 
+
+The prover algorithm is implemented in the \ref bb::SumcheckProver< Flavor > "Sumcheck Prover" class. See its documentation for a more detailed description of methods described below. The Sumcheck Round routine is abstracted into \ref bb::SumcheckProverRound< Flavor > "Sumcheck Prover Round" class, which contains most computational steps.
```

Comment:
of a round of sumcheck

Reply:
It's the entire prover, the link to sumcheck round prover is in the next sentence